src/p_saveg.c

```diff
@@ -973,6 +974,21 @@ static void saveg_read_player_t(player_t *str)
         str->slope = 0;
         str->maxkilldiscount = 0;
     }
+
+    if (saveg_compat > saveg_woof1300)
+    {
+        str->num_visitedlevels = saveg_read32();
+        array_clear(str->visitedlevels);
+        for (int i = 0; i < str->num_visitedlevels; ++i)
+        {
+            array_push(str->visitedlevels, saveg_read32());
```

Comment:
Is there a better way to find out the  "visited levels"? Each UMAPINFO map may have a secret exit, so not sure how to figure out the player's path.

Reply:
I was facing a similar issue when I implemented support for E1M10 in Crispy. 

https://github.com/fabiangreffrath/crispy-doom/blob/edc1e6efbf0cd4af484c113eda2f72b6088b7386/src/doom/wi_stuff.c#L864-L872

My conclusion back then was that you can only really know if the player had made it through a level when leaving it. That is, you need to explicitly keep track of it. This is unambiguous for the IWAD levels and keeping track of the path is sufficient with the `didsecret` variable.

Reply:
> My conclusion back then was that you can only really know if the player had made it through a level when leaving it. That is, you need to explicitly keep track of it.

Agreed, I keep track of the player's path in the `visitedlevels` array and change the saves version.